### PR TITLE
introduce make_boolean_expr

### PR DESCRIPTION
**Pull request by kroening**
- [X] Each commit message has a non-empty body, explaining why the change was made.
- [X] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by smowton — APPROVED** [review]

File: src/util/std_expr.h [hunk 1]
```diff
@@ -4121,6 +4121,9 @@ class false_exprt:public constant_exprt
   }
 };
 
+/// \brief returns true_exprt if given true and false_exprt otherwise
+constant_exprt make_boolean_expr(bool);
```
Comment (tautschnig) [inline src/util/std_expr.h]: Two nit picks:
1) `\brief` is redundant and using a full sentence (starting with an uppercase latter and ending in a full stop) would be nice.
2) I'm wondering whether `std_expr.h` is the right place for this. It's already a way-too-big file, and this doesn't fit particularly well I think. (With the additional thought that `conjunction`, `disjunction` should move as well.)

File: src/util/expr.h [hunk 2]
```diff
@@ -236,6 +236,7 @@ class exprt:public irept
   DEPRECATED(SINCE(2019, 1, 19, "use typecast_exprt() instead"))
   void make_typecast(const typet &_type);
 
+  DEPRECATED(SINCE(2019, 5, 28, "use make_boolean_value(value) instead"))
```
Comment (tautschnig) [inline src/util/expr.h]: `make_boolean_expr` (not `make_boolean_value`)

**Review by tautschnig — APPROVED** [review]
Modulo the comments posted moments ago.

File: src/util/std_expr.h (hunk 1 above)
Comment (kroening) [reply src/util/std_expr.h]: 2. expr_util.h?

Comment (tautschnig) [reply src/util/std_expr.h]: I do think that would be a better place, yes.

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: 9b236c15ce7a8dc94952d8531e4678ecd0420133).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/113455813
Status will be re-evaluated on next push.
Common spurious failures include: the cbmc commit has disappeared in the mean time (e.g. in a force-push); the author is not in the list of contributors (e.g. first-time contributors); compatibility was already broken by an earlier merge.

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: 730928c01fb60bff10bef643408877825fc00e35).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/113461406
Status will be re-evaluated on next push.
Common spurious failures include: the cbmc commit has disappeared in the mean time (e.g. in a force-push); the author is not in the list of contributors (e.g. first-time contributors); compatibility was already broken by an earlier merge.

**Comment by tautschnig** [comment]
@kroening Please rebase now that #4729 is in to make CI happy.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: e60d6e94bc5bd63ed0979689282240cbd1bcdfa9).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/113731033

**Pull request merged**